unionvms-test/src/test/java/eu/europa/ec/fisheries/uvms/docker/validation/activity/ActivityJmsIT.java

```diff
@@ -108,10 +108,17 @@ public void createFishingReport() throws Exception {
 
 		messageHelper.sendMessage("UVMSActivityEvent", reportMessageRequest);
 
+		List<FishingActivityReportDTO> fishingReportsForTripId = getFishingReportsForTripId(newTripId);
+
+		int numberOfFishingActivityReports = fishingReportsForTripId.size();
+
 		// Wait for reports to be persisted
-		Thread.sleep(10000);
+		while (numberOfFishingActivityReports == 0) {
```

Comment:
So much better! Together with test timeout it is definitely more efficient and robust.